### PR TITLE
Add sphinx docu

### PR DESCRIPTION
**Pull request by DunklesArchipel**
***This description is deprecated***

I tried to implement rendering of notebooks to provide nice documentation of the API.

I tested the following approaches:
* https://github.com/greenape/mknotebooks
* https://github.com/danielfrg/mkdocs-jupyter

As far as I understand, neither of them allows rendering of MD files created with jupytext.

The current configuration allows me to render the notebook but without ouptput, even though during the build process mkdocs indicates that the notebook is executed:
` Converting notebook (execute=True): F:\Github\echemdb\website\pages\api\api.ipynb`

For both approaches, I get this error during the build, which seems to be specific for windows, so maybe this works on your system.

```
C:\Miniconda3\envs\echemdb\lib\site-packages\zmq\_future.py:591: RuntimeWarning:

Proactor event loop does not implement add_reader family of methods required for zmq. 
Registering an additional selector thread for add_reader support via tornado. 
Use `asyncio.set_event_loop_policy(WindowsSelectorEventLoopPolicy())` to avoid this warning.
```

### CODE REVIEW

**Comment by saraedum** [comment]
Is it maybe easier to use sphinx instead for this part?

I.e., just generate the documentation for the website Python project with sphinx and put the notebooks in there? Presumably we'd want to link to the actual API from these notebooks more than we actually want to content in the echemdb so maybe that's more natural after all.

We'd get two separate websites of course. One with the actual echemdb and one with the website Python project (which we should eventually rename) but they could quite transparently link back and forth between each other.

**Comment by DunklesArchipel** [comment]
> Is it maybe easier to use sphinx instead for this part?
> 
> I.e., just generate the documentation for the website Python project with sphinx and put the notebooks in there? Presumably we'd want to link to the actual API from these notebooks more than we actually want to content in the echemdb so maybe that's more natural after all.
> 
> We'd get two separate websites of course. One with the actual echemdb and one with the website Python project (which we should eventually rename) but they could quite transparently link back and forth between each other.

I agree. I set up sphinx according to the one in svgdigitizer. At least it builds locally.

As far as I understand we can only host a single site per repository? Or you push one of the sites to a branch of another repo.

In principle, this repo could be called `echemdb` hosting the code and the documentation. The website repo would only contain the build of the MKdocs website. Is that how it could work?

**Comment by saraedum** [comment]
> As far as I understand we can only host a single site per repository? Or you push one of the sites to a branch of another repo.

We can just serve it from a subdirectory, i.e., echemdb.github.io/website/api.

File: doc/conf.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+html_static_path = []
+
+# Add Edit on GitHub links
+# html_context = {
+#     'display_github': True,
+#     'github_user': 'echemdb',
+#     'github_repo': 'website',
+#     'github_version': 'main/doc/',
+# }
```
Comment (DunklesArchipel) [inline doc/conf.py:32]: Linkchecker fails, when new doc files are created, which are not available in the published repo. Hence all "Edit on Github pages" Link checks fail.

I did not find a workaround other than turning it off in any PR that contains new sites in the docu section.

**Comment by DunklesArchipel** [comment]
This PR adds a link to the website, directing to the documentation of the echemdb module and basic API functionality.

I added some pages with a minimum of information on what echemdb can be used for. Besides, there are a number of todos (which are probably far from complete). I would extend #118 with tasks for this purpose, otherwise this PR is getting too large. I would rather like the technical problems solved here.

To view the site locally
```
conda activate echemdb
mkdocs serve
cd doc
make html
python -m http.server 8880 --directory ../generated/website
```
the doc can be updated anytime.

If you update mkdocs, the directory is cleared and you have to rebuild the documentation.

File: doc/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Sphinx Documentation svgdigitizer
```
Comment (saraedum) [inline doc/README.md:1]:
```suggestion
# Sphinx Documentation echemdb
```

File: doc/conf.py (hunk 1 above)
Comment (saraedum) [reply doc/conf.py:32]: You can tell linkchecker to ignore certain URLs. Maybe we should just ignore the ones that go to the edit on github targets?

**Review by saraedum — APPROVED** [review]

**Pull request merged**